[BUG] Workflow typerror during integrations tests

```
tests/integration/test_nvt_tf_inference.py:206: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

root_df =       Store  DayOfWeek  ... StateHoliday_fw  Promo_fw
0       386          2  ...               0         4
1       97...       2  ...               0         7
1023    511          2  ...               0         7

[1024 rows x 75 columns]
workflow_nodes = [<Node + output>], additional_columns = None
capture_dtypes = False

    def _transform_partition(root_df, workflow_nodes, additional_columns=None, capture_dtypes=False):
        """Transforms a single partition by appyling all operators in a WorkflowNode"""
        output = None
    
        for node in workflow_nodes:
            node_input_cols = _get_unique(node.input_schema.column_names)
            node_output_cols = _get_unique(node.output_schema.column_names)
            addl_input_cols = set(node.dependency_columns.names)
    
            # Build input dataframe
            if node.parents_with_dependencies:
                # If there are parents, collect their outputs
                # to build the current node's input
                input_df = None
                seen_columns = None
    
                for parent in node.parents_with_dependencies:
                    parent_output_cols = _get_unique(parent.output_schema.column_names)
                    parent_df = _transform_partition(root_df, [parent], capture_dtypes=capture_dtypes)
                    if input_df is None or not len(input_df):
                        input_df = parent_df[parent_output_cols]
                        seen_columns = set(parent_output_cols)
                    else:
                        new_columns = set(parent_output_cols) - seen_columns
                        input_df = concat_columns([input_df, parent_df[list(new_columns)]])
                        seen_columns.update(new_columns)
    
                # Check for additional input columns that aren't generated by parents
                # and fetch them from the root dataframe
                unseen_columns = set(node.input_schema.column_names) - seen_columns
                addl_input_cols = addl_input_cols.union(unseen_columns)
    
                # TODO: Find a better way to remove dupes
                addl_input_cols = addl_input_cols - set(input_df.columns)
    
                if addl_input_cols:
                    input_df = concat_columns([input_df, root_df[list(addl_input_cols)]])
            else:
                # If there are no parents, this is an input node,
                # so pull columns directly from root df
                input_df = root_df[node_input_cols + list(addl_input_cols)]
    
            # Compute the node's output
            if node.op:
                try:
                    # use input_columns to ensure correct grouping (subgroups)
                    selection = node.input_columns.resolve(node.input_schema)
                    output_df = node.op.transform(selection, input_df)
    
                    # Update or validate output_df dtypes
                    for col_name, output_col_schema in node.output_schema.column_schemas.items():
                        col_series = output_df[col_name]
                        col_dtype = col_series.dtype
                        is_list = is_list_dtype(col_series)
    
                        if is_list:
                            col_dtype = list_val_dtype(col_series)
    
                        output_df_schema = output_col_schema.with_dtype(
                            col_dtype, is_list=is_list, is_ragged=is_list
                        )
    
                        if capture_dtypes:
                            node.output_schema.column_schemas[col_name] = output_df_schema
                        elif len(output_df):
                            if output_col_schema.dtype != output_df_schema.dtype:
>                               raise TypeError(
                                    f"Improperly matched output dtypes detected in {col_name},"
                                    f" {output_col_schema.dtype} and {output_df_schema.dtype}"
                                )
E                               TypeError: Improperly matched output dtypes detected in CompetitionDistance, float32 and float64

nvtabular/workflow/workflow.py:485: TypeError
```

--- a/examples/tabular-data-rossmann/03-Training-with-TF.ipynb
+++ b/examples/tabular-data-rossmann/03-Training-with-TF.ipynb
@@ -345,7 +345,7 @@
     "# Just concatenating continuous, so can use a list\n",
     "continuous_inputs = []\n",
     "for column_name in CONTINUOUS_COLUMNS:\n",
-    "    continuous_inputs.append(tf.keras.Input(name=column_name, shape=(1,), dtype=tf.float32))\n",
+    "    continuous_inputs.append(tf.keras.Input(name=column_name, shape=(1,), dtype=tf.float64))\n",
     "continuous_embedding_layer = tf.keras.layers.Concatenate(axis=1)\n",
     "continuous_x = continuous_embedding_layer(continuous_inputs)\n",
     "continuous_x = tf.keras.layers.BatchNormalization(epsilon=1e-5, momentum=0.1)(continuous_x)\n",
@@ -542,7 +542,7 @@
  "metadata": {
   "file_extension": ".py",
   "kernelspec": {
-   "display_name": "Python 3",
+   "display_name": "Python 3 (ipykernel)",
    "language": "python",
    "name": "python3"
   },
@@ -556,7 +556,7 @@
    "name": "python",
    "nbconvert_exporter": "python",
    "pygments_lexer": "ipython3",
-   "version": "3.8.5"
+   "version": "3.8.10"
   },
   "mimetype": "text/x-python",
   "name": "python",

